src/resource_clients/key_value_store.ts

```diff
@@ -177,7 +177,8 @@ export class KeyValueStoreClient extends ResourceClient {
         );
 
         const { key } = record;
-        let { value, contentType } = record;
+        // eslint-disable-next-line prefer-const
+        let { value, contentType, timeoutSecs, doNotRetryTimeouts } = record;
```

Comment:
(nit) This is not really part of the record. Wouldn't it be cleaner to pass this inside an optional `options` param?

Reply:
I knew you won't like this bit 🤣

Reply:
I'll refactor it here, but in crawlee I'd rather keep a single options object, feels weird to have two options objects next to each other.

https://github.com/apify/crawlee/blob/master/packages/core/src/storages/key_value_store.ts#L328